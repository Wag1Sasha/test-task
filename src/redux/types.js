export const ADD_TAG = 'TAGS/ADD_TAG'
export const DELETE_LAST_TAG = 'TAGS/DELETE_LAST_TAG'