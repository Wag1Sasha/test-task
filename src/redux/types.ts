export const EVENT_SUBMIT = 'TAGS/EVENT_SUBMIT';
export const DELETE_TAG = 'TAGS/DELETE_TAG';
export const REQUEST_PHOTOS_BY_VALUE = 'PHOTOS/REQUEST_PHOTOS_BY_VALUE';
export const ASYNC_FETCH_PHOTOS = 'ASYNC_FETCH_PHOTOS';
export const ADD_PHOTO_TO_STORE = 'PHOTOS/ADD_PHOTO_TO_STORE';