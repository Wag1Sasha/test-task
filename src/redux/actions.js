import { EVENT_SUBMIT,DELETE_TAG, ADD_PHOTO_TO_STORE, ASYNC_FETCH_PHOTOS } from './types';

export const eventSubmit = (tags) => ({
    type: EVENT_SUBMIT,
    payload: {
      tags: Object.values(tags).filter(word => word.length > 0)
    }
});

export const deleteTag = (i) =>({
    type:DELETE_TAG,
    payload:i
});


export const addPhotoToStore = (photos) => ({
    type: ADD_PHOTO_TO_STORE,
    payload: photos,
});


export const asyncFetchPhotos = (url) => ({
  type: ASYNC_FETCH_PHOTOS,
  url
});