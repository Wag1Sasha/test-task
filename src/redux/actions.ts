import { EVENT_SUBMIT, DELETE_TAG, ADD_PHOTO_TO_STORE, ASYNC_FETCH_PHOTOS, DRAG_N_DROP } from './types';

export const eventSubmit = (tags: TagPropsType) => ({
  type: EVENT_SUBMIT,
  payload: tags,
});

export const dragNDrop = (tags: TagPropsType[]) => ({
  type: DRAG_N_DROP,
  payload: tags,
});

export const deleteTag = (index: number): TagsActionTypes => ({
  type: DELETE_TAG,
  payload: index,
});

export const addPhotoToStore = (photos: IPropsHits) => ({
  type: ADD_PHOTO_TO_STORE,
  payload: photos,
});

export const asyncFetchPhotos = (url: string) => ({
  type: ASYNC_FETCH_PHOTOS,
  url,
});
