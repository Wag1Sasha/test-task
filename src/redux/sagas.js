import { takeEvery, put, call } from 'redux-saga/effects';
import { ASYNC_FETCH_PHOTOS} from './types';
import {addPhotoToStore} from './actions';

export function* sagaWarcher() {
  yield takeEvery(ASYNC_FETCH_PHOTOS, sagaWorker)
};

function* sagaWorker({url}) {
  const payload = yield call(requestPhotos,url)
  yield put(addPhotoToStore(payload.hits))
};


const requestPhotos =async(u)=>{
  const res = await fetch(`https://pixabay.com/api/?key=19776687-7c6cc257a41e8830f998f821c&q=${u}`)
  const data = await res.json();
  return data;
};