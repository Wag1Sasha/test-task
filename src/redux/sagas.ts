import { takeEvery, put, call } from 'redux-saga/effects';
import { ASYNC_FETCH_PHOTOS } from './types';
import { addPhotoToStore, asyncFetchPhotos } from './actions';

export function* sagaWarcher() {
  yield takeEvery(ASYNC_FETCH_PHOTOS, sagaWorker);
}

function* sagaWorker(payload: ReturnType<typeof asyncFetchPhotos>) {
  const data = yield call(requestPhotos, payload.url);
  yield put(addPhotoToStore(data.hits));
}

const requestPhotos = async (u: string) => {
  const res = await fetch(`https://pixabay.com/api/?key=19776687-7c6cc257a41e8830f998f821c&q=${u}`);
  return await res.json();
};
