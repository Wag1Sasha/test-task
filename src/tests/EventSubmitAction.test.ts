import configureStore from 'redux-mock-store';

import * as selectActions from '../redux/actions';

describe('Event submit action-creator', () => {
  const mockStore = configureStore();
  const store = mockStore();
  const tag = { text: 'text', id: 'text' };
  beforeEach(() => {
    store.clearActions();
  });
  it('Action-creator works correct', () => {
    const expectedActions = [
      {
        type: 'TAGS/EVENT_SUBMIT',
        payload: { text: 'text', id: 'text' },
      },
    ];
    store.dispatch(selectActions.eventSubmit(tag));
    expect(store.getActions()).toEqual(expectedActions);
  });
  it('Render action-creator', () => {
    store.dispatch(selectActions.eventSubmit(tag));
    expect(store.getActions()).toMatchSnapshot();
  });
});
