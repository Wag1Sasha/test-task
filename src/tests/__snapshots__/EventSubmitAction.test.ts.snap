// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event submit action-creator Render action-creator 1`] = `
Array [
  Object {
    "payload": Object {
      "id": "text",
      "text": "text",
    },
    "type": "TAGS/EVENT_SUBMIT",
  },
]
`;
